diff --git print_handler_gtk.cc print_handler_gtk.cc
--- print_handler_gtk.cc
+++ print_handler_gtk.cc
@@ -1,9 +1,12 @@
+// COPIED from upstream "cef/tests/cefclient/browser/" directory
+// with minor modifications. See the .patch file in current directory.
+
 // Copyright (c) 2014 The Chromium Embedded Framework Authors.
 // Portions Copyright (c) 2012 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "tests/cefclient/browser/print_handler_gtk.h"
+#include "print_handler_gtk.h"
 
 #include <gtk/gtk.h>
 #include <gtk/gtkunixprint.h>
@@ -15,10 +18,10 @@
 #include "include/base/cef_logging.h"
 #include "include/base/cef_macros.h"
 #include "include/wrapper/cef_helpers.h"
-#include "tests/cefclient/browser/root_window.h"
-#include "tests/cefclient/browser/util_gtk.h"
+#include "include/wrapper/cef_closure_task.h"
 
-namespace client {
+#include "client_handler/util_gtk.h"
+#include "client_handler/x11.h"
 
 namespace {
 
@@ -280,12 +283,7 @@
 // Returns the GtkWindow* for the browser. Will return nullptr when using the
 // Views framework.
 GtkWindow* GetWindow(CefRefPtr<CefBrowser> browser) {
-  scoped_refptr<RootWindow> root_window =
-      RootWindow::GetForBrowser(browser->GetIdentifier());
-  if (root_window) {
-    return GTK_WINDOW(root_window->GetWindowHandle());
-  }
-  return nullptr;
+  return CefBrowser_GetGtkWindow(browser);
 }
 
 void GetWindowAndContinue(CefRefPtr<CefBrowser> browser,
@@ -645,5 +643,3 @@
 
   return CefSize(width * device_units_per_inch, height * device_units_per_inch);
 }
-
-}  // namespace client
--- print_handler_gtk.h
+++ print_handler_gtk.h
@@ -1,3 +1,6 @@
+// COPIED from upstream "cef/tests/cefclient/browser/" directory
+// with minor modifications. See the .patch file in current directory.
+
 // Copyright (c) 2014 The Chromium Embedded Framework Authors.
 // Portions Copyright (c) 2012 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
@@ -11,8 +14,6 @@
 
 #include "include/cef_print_handler.h"
 
-namespace client {
-
 class ClientPrintHandlerGtk : public CefPrintHandler {
  public:
   ClientPrintHandlerGtk();
@@ -45,6 +46,4 @@
   IMPLEMENT_REFCOUNTING(ClientPrintHandlerGtk);
 };
 
-}  // namespace client
-
 #endif  // CEF_TESTS_CEFCLIENT_BROWSER_PRINT_HANDLER_GTK_H_
