diff --git dialog_handler_gtk.cc dialog_handler_gtk.cc
--- dialog_handler_gtk.cc
+++ dialog_handler_gtk.cc
@@ -1,9 +1,10 @@
+// COPIED from upstream "cef/tests/cefclient/browser/" directory
+// with minor modifications. See the .patch file in current directory.
+
 // Copyright (c) 2015 The Chromium Embedded Framework Authors. All rights
 // reserved. Use of this source code is governed by a BSD-style license that
 // can be found in the LICENSE file.
 
-#include "tests/cefclient/browser/dialog_handler_gtk.h"
-
 #include <libgen.h>
 #include <sys/stat.h>
 
@@ -11,11 +12,11 @@
 #include "include/cef_browser.h"
 #include "include/cef_parser.h"
 #include "include/wrapper/cef_helpers.h"
-#include "tests/cefclient/browser/root_window.h"
-#include "tests/cefclient/browser/util_gtk.h"
-#include "tests/shared/common/string_util.h"
 
-namespace client {
+#include "include/base/cef_logging.h"
+#include "dialog_handler_gtk.h"
+#include "util_gtk.h"
+#include "x11.h"
 
 namespace {
 
@@ -130,17 +131,7 @@
 
 GtkWindow* GetWindow(CefRefPtr<CefBrowser> browser) {
   REQUIRE_MAIN_THREAD();
-  scoped_refptr<RootWindow> root_window =
-      RootWindow::GetForBrowser(browser->GetIdentifier());
-  if (root_window) {
-    GtkWidget* window = root_window->GetWindowHandle();
-    DCHECK(window);
-    if (!window) {
-      LOG(ERROR) << "No GtkWindow for browser";
-    }
-    return GTK_WINDOW(window);
-  }
-  return nullptr;
+  return CefBrowser_GetGtkWindow(browser);
 }
 
 // Returns true if |accept_filters| contains a MIME type value without matching
@@ -379,19 +370,19 @@
     case JSDIALOGTYPE_ALERT:
       buttons = GTK_BUTTONS_NONE;
       gtk_message_type = GTK_MESSAGE_WARNING;
-      title = "JavaScript Alert";
+      title = "Alert";
       break;
 
     case JSDIALOGTYPE_CONFIRM:
       buttons = GTK_BUTTONS_CANCEL;
       gtk_message_type = GTK_MESSAGE_QUESTION;
-      title = "JavaScript Confirm";
+      title = "Confirm";
       break;
 
     case JSDIALOGTYPE_PROMPT:
       buttons = GTK_BUTTONS_CANCEL;
       gtk_message_type = GTK_MESSAGE_QUESTION;
-      title = "JavaScript Prompt";
+      title = "Prompt";
       break;
 
     case JSDIALOGTYPE_NUM_VALUES:
@@ -402,8 +393,8 @@
   js_dialog_callback_ = params.callback;
 
   if (!params.origin_url.empty()) {
-    title += " - ";
-    title += CefFormatUrlForSecurityDisplay(params.origin_url).ToString();
+    // title += " - ";
+    // title += CefFormatUrlForSecurityDisplay(params.origin_url).ToString();
   }
 
   gtk_dialog_ = gtk_message_dialog_new(GTK_WINDOW(window), GTK_DIALOG_MODAL,
@@ -476,5 +467,3 @@
               base::BindOnce(&ClientDialogHandlerGtk::OnResetDialogState,
                              handler, nullptr));
 }
-
-}  // namespace client
--- dialog_handler_gtk.h
+++ dialog_handler_gtk.h
@@ -1,3 +1,6 @@
+// COPIED from upstream "cef/tests/cefclient/browser/" directory
+// with minor modifications. See the .patch file in current directory.
+
 // Copyright (c) 2015 The Chromium Embedded Framework Authors. All rights
 // reserved. Use of this source code is governed by a BSD-style license that
 // can be found in the LICENSE file.
@@ -12,8 +15,6 @@
 #include "include/cef_dialog_handler.h"
 #include "include/cef_jsdialog_handler.h"
 
-namespace client {
-
 class ClientDialogHandlerGtk : public CefDialogHandler,
                                public CefJSDialogHandler {
  public:
@@ -83,6 +84,4 @@
   IMPLEMENT_REFCOUNTING(ClientDialogHandlerGtk);
 };
 
-}  // namespace client
-
 #endif  // CEF_TESTS_CEFCLIENT_BROWSER_DIALOG_HANDLER_GTK_H_
